src/index.ts

```diff
@@ -1,4 +1,36 @@
-import { run } from 'probot';
+import { Server, Probot } from 'probot';
 import app from './app';
+import { register } from 'prom-client';
 
-run(app);
+async function startServer() {
+  const probotDefault = Probot.defaults({
+    appId: process.env.APP_ID,
+    privateKey: process.env.PRIVATE_KEY,
+    secret: process.env.WEBHOOK_SECRET,
+  });
+
+  const server =
+    process.env.NODE_ENV == 'production'
+      ? new Server({
+          Probot: probotDefault,
+        })
+      : new Server({
+          webhookProxy: process.env.WEBHOOK_PROXY_URL,
```

Comment:
This does not start the webhook proxy by itself. It doesn't instantiate the `smee-client` the same way as `run()` call does. I think you're missing something like:

```ts
const smee = process.env.WEBHOOK_PROXY_URL
  ? undefined
  : new SmeeClient({
      source: process.env.WEBHOOK_PROXY_URL || '',
      target: 'http://localhost:3000/events',
      logger: console,
    });

const events = smee?.start();
...
process.env.WEBHOOK_PROXY_URL &&
  [
    `exit`,
    `SIGINT`,
    `SIGUSR1`,
    `SIGUSR2`,
    `uncaughtException`,
    `SIGTERM`,
  ].forEach((_) => {
    events?.close();
  });
```

to work aroud this.